### PR TITLE
Khp2 3130

### PR DESCRIPTION
**Pull request by njorocs**

### CODE REVIEW

File: omod/src/main/webapp/fragments/kpClient/clientSummary.gsp [hunk 1]
```diff
@@ -9,22 +9,17 @@
 		<table>
 			<tr>
 				<th rowspan="2">Status In program</th>
-				<th rowspan="2">ART Status</th>
+				<th rowspan="2">ART Status/Latest viral load</th>
```
Comment (makombe) [inline omod/src/main/webapp/fragments/kpClient/clientSummary.gsp:12]: Why combine Art status and Vl ? I think they need to be separated

File: omod/src/main/webapp/fragments/kpClient/clientSummary.gsp [hunk 2]
```diff
@@ -9,22 +9,17 @@
...
-
 			</tr>
 	<tr></tr>
 
 			<tr>
-			 <td width="80px">${encounters.statusInProgram}</td>
-			<td>${nextVisitDate}</td>
-			<td>${nextVisitDate}</td>
+			 <td width="140px">${statusInProgram}</td>
+			<td>${artStatus}</td>
```
Comment (makombe) [inline omod/src/main/webapp/fragments/kpClient/clientSummary.gsp:19]: You are only binding artStatus, where is the Vl?

Comment (njorocs) [reply omod/src/main/webapp/fragments/kpClient/clientSummary.gsp:19]: artStatus is a composition of VL and ART status. Let me rename the variable appropriately

File: omod/src/main/webapp/fragments/kpClient/clientSummary.gsp (hunk 1 above)
Comment (njorocs) [reply omod/src/main/webapp/fragments/kpClient/clientSummary.gsp:12]: The requirement expects both in one section.

File: omod/src/main/java/org/openmrs/module/kenyakeypop/fragment/controller/kpClient/ClientSummaryFragmentController.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+		/*	CalculationResult returnVisitResults = EmrCalculationUtils.evaluateForPatient(LastReturnVisitDateCalculation.class, null, patient);
+			if(returnVisitResults != null){
+				patientSummary.setNextAppointmentDate(formatDate((Date) returnVisitResults.getValue()));
+			}
+			else {
+				patientSummary.setNextAppointmentDate("");
+			}*/
+		
+		//String encDate = e != null ? DATE_FORMAT.format(e.getEncounterDatetime()) : "";
+
```
[the remark this answers is not in the document]
Comment (njorocs) [reply omod/src/main/java/org/openmrs/module/kenyakeypop/fragment/controller/kpClient/ClientSummaryFragmentController.java]: Sending a fresh PR. This was taken care in a later commit.

File: omod/src/main/java/org/openmrs/module/kenyakeypop/fragment/controller/kpClient/ClientSummaryFragmentController.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	public static SimpleObject buildEncounterData(Set<Obs> obsList) {
+		/*			CalculationResultMap nextAppointmentMap = Calculations.lastObs(Dictionary.getConcept(Dictionary.RETURN_VISIT_DATE), cohort, context);
+					Obs nextOfVisitObs = EmrCalculationUtils.obsResultForPatient(nextAppointmentMap, ptId);
+					Date notAlostToFollowPatient = DateUtil.adjustDate(nextOfVisitObs.getValueDatetime(), 90, DurationUnit.DAYS);*/
+		
+		int STATUS_IN_PROGRAM = 161641;
+		
+		String statusInProgram = null;
+		/*String nextAppointmentDate = null;
+		String frequentedHotspot = null;
```
[the remark this answers is not in the document]
Comment (njorocs) [reply omod/src/main/java/org/openmrs/module/kenyakeypop/fragment/controller/kpClient/ClientSummaryFragmentController.java]: Sending a fresh PR. This was taken care in a later commit.

File: omod/src/main/java/org/openmrs/module/kenyakeypop/fragment/controller/kpClient/ClientSummaryFragmentController.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+			simplifiedEncData.add(o);
+		}
+		
+		CalculationResult nextAppointmentResults = EmrCalculationUtils.evaluateForPatient(
+		    NextAppointmentDateCalculation.class, null, patient);
+		String nextVisitDate;
+		
+		nextVisitDate = this.formatDate((Date) nextAppointmentResults.getValue());
+		/*	if (nextAppointmentResults.isEmpty()) {
+				//System.out.println("=======================" + nextAppointmentDate.getValue());
```
[the remark this answers is not in the document]
Comment (njorocs) [reply omod/src/main/java/org/openmrs/module/kenyakeypop/fragment/controller/kpClient/ClientSummaryFragmentController.java]: Sending a fresh PR. This was taken care in a later commit.

File: api/src/main/java/org/openmrs/module/kenyakeypop/calculation/library/kp/NextAppointmentDateCalculation.java [hunk 6]
```diff
@@ -3,50 +3,62 @@
...
 import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.SimpleResult;
 import org.openmrs.module.kenyacore.calculation.AbstractPatientCalculation;
-import org.openmrs.module.kenyacore.calculation.BooleanResult;
-import org.openmrs.module.kenyacore.calculation.Calculations;
-import org.openmrs.module.kenyaemr.Dictionary;
-import org.openmrs.module.kenyaemr.calculation.EmrCalculationUtils;
-import org.openmrs.module.reporting.common.DateUtil;
-import org.openmrs.module.reporting.common.DurationUnit;
+import org.openmrs.module.kenyaemr.metadata.HivMetadata;
```
[the remark this answers is not in the document]
Comment (njorocs) [reply api/src/main/java/org/openmrs/module/kenyakeypop/calculation/library/kp/NextAppointmentDateCalculation.java]: File removed in a later commit. Sending a fresh PR

File: api/src/main/java/org/openmrs/module/kenyakeypop/calculation/library/kp/NextAppointmentDateCalculation.java [hunk 7]
```diff
@@ -3,50 +3,62 @@
...
+	@Override
+	public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> map,
+	        PatientCalculationContext context) {
+		
+		Integer nextAppointmentDate = 5096;
+		CalculationResultMap ret = new CalculationResultMap();
+		
+		Form kpClinicalVisit = MetadataUtils.existing(Form.class, KpMetadata._Form.KP_CLINICAL_VISIT_FORM);
+		
+		EncounterType kpFollowup = MetadataUtils.existing(EncounterType.class, HivMetadata._EncounterType.HIV_CONSULTATION);
```
[the remark this answers is not in the document]
Comment (njorocs) [reply api/src/main/java/org/openmrs/module/kenyakeypop/calculation/library/kp/NextAppointmentDateCalculation.java]: File removed in a later commit. Sending a fresh PR

**Pull request closed without merging**